src/sas/sascalc/dataloader/readers/sesans_reader.py

```diff
@@ -17,158 +17,163 @@
+                    x = data[:, 0]
+                    dx = data[:, 3]
+                    lam = data[:, 4]
+                    dlam = data[:, 5]
+                    y = data[:, 1]
+                    dy = data[:, 2]
+
+                    lam_unit = self._header_fetch(headers, "wavelength")
+                    if lam_unit == "AA":
+                        lam_unit = "A"
+
+                    x, x_unit = self._unit_conversion(
+                        x, lam_unit,
+                        self._fetch_unit(headers, "spin echo length"))
+                    dx, dx_unit = self._unit_conversion(
+                        dx, lam_unit,
+                        self._fetch_unit(headers, "error SEL"))
+                    dlam, dlam_unit = self._unit_conversion(
+                        dlam, lam_unit,
+                        self._fetch_unit(headers, "error wavelength"))
+                    y_unit = r'\AA^{-2} cm^{-1}'
+
+                    output = Data1D(x=x, y=y, lam=lam, dy=dy, dx=dx, dlam=dlam,
+                                    isSesans=True)
+                    self.filename = output.filename = basename
+                    output.xaxis(r"\rm{z}", x_unit)
+                    # Adjust label to ln P/(lam^2 t), remove lam column refs
+                    output.yaxis(r"\rm{ln(P)/(t \lambda^2)}", y_unit)
+                    # Store loading process information
+                    output.meta_data['loader'] = self.type_name
+                    output.sample.name = params["Sample"]
+                    output.sample.ID = params["DataFileTitle"]
+
+                    output.sample.zacceptance = (
+                        float(self._header_fetch(params, "Q_zmax")),
+                        self._fetch_unit(params, "Q_zmax"))
+
+                    output.sample.yacceptance = (
+                        float(self._header_fetch(params, "Q_ymax")),
+                        self._fetch_unit(params, "Q_ymax"))
 
                 if len(output.x) < 1:
-                    raise RuntimeError, "%s is empty" % path
+                    raise RuntimeError("%s is empty" % path)
                 return output
 
         else:
-            raise RuntimeError, "%s is not a file" % path
+            raise RuntimeError("%s is not a file" % path)
         return None
 
-    def _unit_conversion(self, value, value_unit, default_unit):
-        if has_converter == True and value_unit != default_unit:
+    @staticmethod
+    def _unit_conversion(value, value_unit, default_unit):
+        """
+        Performs unit conversion on a measurement.
+
+        :param value: The magnitude of the measurement
+        :type value: Number
+        :param value_unit: a string containing the final desired unit
+        :type value_unit: String
+        :param default_unit: a string containing the units of the original measurement
+        :type default_unit: String
+        :return: The magnitude of the measurement in the new units
+        """
```

Comment:
We ought to be using consistent markup across the project for function docs (though something is definitely better than nothing!)  Maybe add a ticket if there is not already one there?

For types I suggest using pep 484 style type hints embedded in the comments:

    # type: (float, str, str) -> float

I've been using "*parameter name* description" for a description of the inputs to the function.  I haven't tried rendering the above format with sphinx and automod.